cdap-security/src/main/java/io/cdap/cdap/security/tools/TokenGeneratorTool.java

```diff
@@ -0,0 +1,120 @@
+import com.google.inject.Key;
+import com.google.inject.TypeLiteral;
+import io.cdap.cdap.api.common.Bytes;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.guice.ConfigModule;
+import io.cdap.cdap.common.guice.IOModule;
+import io.cdap.cdap.common.guice.InMemoryDiscoveryModule;
+import io.cdap.cdap.common.io.Codec;
+import io.cdap.cdap.security.auth.AbstractKeyManager;
+import io.cdap.cdap.security.auth.AccessToken;
+import io.cdap.cdap.security.auth.FileBasedKeyManager;
+import io.cdap.cdap.security.auth.TokenManager;
+import io.cdap.cdap.security.auth.UserIdentity;
+import io.cdap.cdap.security.guice.CoreSecurityRuntimeModule;
+import org.apache.commons.cli.BasicParser;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Base64;
+import java.util.Collections;
+
+/**
+ * A command line tool for generating access tokens.
+ */
+public class TokenGeneratorTool {
+
+  private void doMain(String[] args) throws ParseException, IOException {
+    Options options = new Options()
+      .addOption(new Option("h", "help", false, "Print this usage message."))
+      .addOption(new Option("u", "username", true, "Username/principal"));
+
+    CommandLineParser parser = new BasicParser();
+    CommandLine commandLine = parser.parse(options, args);
+
+    if (commandLine.hasOption("h") || commandLine.getOptions().length == 0) {
+      printUsage(options);
+      return;
+    }
+
+    if (commandLine.hasOption("u")) {
+      String username = commandLine.getOptionValue("u");
+      printAccessToken(username);
+      return;
+    }
+
+    printUsage(options);
+  }
+
+  @VisibleForTesting
+  void printAccessToken(String username) throws IOException {
+    // Suppress logging from CDAP classes
+    ((Logger) LoggerFactory.getLogger(TokenManager.class)).setLevel(Level.ERROR);
+    ((Logger) LoggerFactory.getLogger(FileBasedKeyManager.class)).setLevel(Level.ERROR);
+    ((Logger) LoggerFactory.getLogger(AbstractKeyManager.class)).setLevel(Level.ERROR);
+
+    CConfiguration cConf = CConfiguration.create();
+    Injector injector = Guice.createInjector(
+      new ConfigModule(cConf),
+      new IOModule(),
+      new InMemoryDiscoveryModule(),
+      CoreSecurityRuntimeModule.getDistributedModule(cConf));
+    TokenManager tokenManager = injector.getInstance(TokenManager.class);
+    tokenManager.startUp();
+
+    long tokenExpiration = cConf.getLong(Constants.Security.TOKEN_EXPIRATION);
+    long issueTime = System.currentTimeMillis();
+    long expireTime = issueTime + tokenExpiration;
+    UserIdentity userIdentity = new UserIdentity(username, UserIdentity.IdentifierType.EXTERNAL,
```

Comment:
nit: although INTERNAL identifier type is not supported at the moment, we may want to also make the identifier type configurable via an option.

Reply:
will fix